src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1436,7 +1436,7 @@ public String create(final String path, byte data[], List<ACL> acl,
         request.setData(data);
         request.setFlags(createMode.toFlag());
         request.setPath(serverPath);
-        if (acl != null && acl.size() == 0) {
+        if (acl == null || acl.isEmpty() || acl.contains(null)) {
```

Comment:
+1 on this. We can throw IllegalArgumentException instead of KeeperException (which would change API signature) on async methods. It's good to catch the bad arguments at client side.